sentenai/api/stream.py

```diff
@@ -0,0 +1,512 @@
+        """Get event as JSON.
+
+        Arguments:
+           eid    -- A unique ID corresponding to an event stored within
+                     the stream.
+        """
+        return self._client.get(self, id)
+    _get = get
+
+    def read(self, id):
+        """Get event as JSON.
+
+        Arguments:
+           eid    -- A unique ID corresponding to an event stored within
+                     the stream.
+        """
+        k = self._client.get(self, id)
+        return Event(self._client, self, data=k['event'], id=k['id'], ts=cts(k['ts']), saved=True)
+    _read = read
+
+    def fstats(self, field, start=None, end=None):
+        """Get stats for a given numeric field.
+
+           Arguments:
+           field  -- A dotted field name for a numeric field in the stream.
+           start  -- Optional argument indicating start time in stream for calculations.
+           end    -- Optional argument indicating end time in stream for calculations.
+        """
+        return self._client.field_stats(self, field, start, end)
+    _fstats = fstats
+
+    def describe(self, field, start=None, end=None):
+        """Describe a given numeric field.
+
+           Arguments:
+           field  -- A dotted field name for a numeric field in the stream.
+           start  -- Optional argument indicating start time in stream for calculations.
+           end    -- Optional argument indicating end time in stream for calculations.
+        """
+        x = self._client.field_stats(self, field, start, end)
+        if x.get('categorical'):
+            print("count\t{count}\nunique\t{unique}\ntop\t{top}\nfreq\t{freq}".format(**x['categorical']))
+        else:
+            p = x['numerical']
+            print("count\t{}\nmean\t{:.2f}\nstd\t{:.2f}\nmin\t{}\n25%\t{}\n50%\t{}\n75%\t{}\nmax\t{}".format(
+                p['count'], p['mean'], p['std'], p['min'], p.get('25%'), p.get('50%'), p.get('75%'), p['max']))
+    _describe = describe
+
+    def unique(self, field):
+        """Get unique values for a given field.
+
+           Arguments:
+           field  -- A dotted field name for a numeric field in the stream.
+           start  -- Optional argument indicating start time in stream for calculations.
+           end    -- Optional argument indicating end time in stream for calculations.
+        """
+        return self._client.unique(self, field)
+    _unique = unique
+
+    def put(self, event, id=None, timestamp=None, duration=None):
+        """Put a new event into this stream.
+
+        Arguments:
+           event     -- A JSON-serializable dictionary containing an
+                        event's data
+           id        -- A user-specified id for the event that is unique to
+                        this stream (optional)
+           timestamp -- A user-specified datetime object representing the
+                        time of the event. (optional)
+        """
+        return self._client.put(self, event, id, timestamp)
```

Comment:
this `put` method takes a `duration` but doesn't use it. `Event#update` tries to pass a `duration` through. does something need to be wired up here or should we strip it out?